scipy/sparse/linalg/interface.py

```diff
@@ -338,6 +340,56 @@ def matmat(self, X):
 
         return Y
 
+    def rmatmat(self, X):
+        """Adjoint matrix-matrix multiplication.
+
+        Performs the operation y = A^H * x where A is an MxN linear
+        operator and x is a column vector or 1-d array, or 2-d array.
+        The default implementation defers to the adjoint.
+
+        Parameters
+        ----------
+        x : {matrix, ndarray}
+            A matrix or 2D array.
+
+        Returns
+        -------
+        y : {matrix, ndarray}
+            A matrix or 2D array depending on the type of the input.
+
+        Notes
+        -----
+        This rmatmat wraps the user-specified rmatmat routine or overridden
+        _rmatmat method to ensure that y has the correct shape and type.
+
+        """
+
+        X = np.asanyarray(X)
+
+        if X.ndim != 2:
+            raise ValueError('expected 2-d ndarray or matrix, not %d-d'
+                             % X.ndim)
+
+        M, _ = self.shape
+
+        if X.shape[0] != M:
+            raise ValueError('dimension mismatch: %r, %r'
+                             % (self.shape, X.shape))
+
+        Y = self._rmatmat(X)
```

Comment:
Silly nitpick: I again know this is just mimicking the surrounding code, but the large amount of blank lines isn't very Pythonic... e.g. PEP8 has this to say:

"Use blank lines in functions, sparingly, to indicate logical sections."

(See https://www.python.org/dev/peps/pep-0008/#blank-lines)

I'd rather see something like

```python
Y = self._rmatmat(X)
if isinstance(Y, np.matrix):
    Y = asmatrix(Y)
return Y
```

Reply:
done